internal/host/plugin/repository_host_catalog.go

```diff
@@ -145,6 +148,299 @@ func (r *Repository) CreateCatalog(ctx context.Context, c *HostCatalog, _ ...Opt
 	return newHostCatalog, plg, nil
 }
 
+// UpdateCatalog updates the repository entry for c.PublicId with the
+// values in c for the fields listed in fieldMask. It returns a new
+// HostCatalog containing the updated values and a count of the number of
+// records updated. c is not changed.
+//
+// c must contain a valid PublicId. c.Name, c.Description, and
+// c.Attributes can be updated; if c.Secrets is present, its contents
+// are sent to the plugin (along with any other changes, see below)
+// before the update is sent to the database.
+//
+// An attribute of c will be set to NULL in the database if the
+// attribute in c is the zero value and it is included in fieldMask.
+// Note that this does not apply to c.Attributes - a null
+// c.Attributes is a no-op for modifications. Rather, if fields need
+// to be reset, its field in c.Attributes should individually set to
+// null.
+//
+// Updates are sent to OnUpdateCatalog with a full copy of both the
+// current catalog, and the state of the new catalog should it be
+// updated, along with any secrets included in the new request. This
+// request may alter the returned persisted state. Update of the
+// record in the database is aborted if this call fails.
+func (r *Repository) UpdateCatalog(ctx context.Context, c *HostCatalog, version uint32, fieldMask []string, _ ...Option) (*HostCatalog, int, error) {
```

Comment:
Can you make this UpdateCatalog call also return the *hostplugin.Plugin like CreateCatalog does?  This is so the response can include the plugin info for the specific resource being updated.

Reply:
Done!